docs/sdks/uid2-sdk-ref-java.md

```diff
@@ -102,6 +103,183 @@ Available information returned through the SDK is outlined in the following tabl
+      ```
+   3. Determine if a refresh is needed:
+
+      ```java
+      if (identity.isDueForRefresh()) {..}
+      ```
+4. If needed, refresh the token and associated values:
+ 
+   ```java
+   TokenRefreshResponse tokenRefreshResponse = publisherUid2Client.refreshToken(identity);
+   ```
+ 
+5. Store `tokenRefreshResponse.getIdentityJsonString()` in the user's session.
+
+   If the user has opted out, this method returns `null`, indicating that the user's identity should be removed from the session. To confirm optout, you can use the `tokenRefreshResponse.isOptout()` function.
+
+### Advanced Usage
+
+1. Create an instance of `PublisherUid2Helper` as an instance variable:
+
+    ```java
+    private final PublisherUid2Helper publisherUid2Helper = new PublisherUid2Helper(UID2_SECRET_KEY);
+    ```
+2. Call a function that takes the user's email address or phone number as input and creates a secure request data envelope. See [Encrypting requests](../getting-started/gs-encryption-decryption.md#encrypting-requests). The following example uses an email address:
+
+    ```java
+    EnvelopeV2 envelope = publisherUid2Helper.createEnvelopeForTokenGenerateRequest(TokenGenerateInput.fromEmail(emailAddress).doNotGenerateTokensForOptedOut());
+    ```
+3. Using an HTTP client library of your choice, post this envelope to the [POST token/generate](../endpoints/post-token-generate.md) endpoint, including headers and body:
+   1. Headers: Depending on your HTTP library, this might look something like the following:  
+    
+      `.putHeader("Authorization", "Bearer " + UID2_API_KEY)`  
+      `.putHeader("X-UID2-Client-Version", PublisherUid2Helper.getVersionHeader())`
+   2. Body: `envelope.getEnvelope()`
+   :::important
+   - Be sure to call this endpoint only when you have obtained legal basis to convert the user’s [directly identifying information (DII)](../ref-info/glossary-uid.md#gl-dii) to UID2 tokens for targeted advertising.
+
+   - Always apply `doNotGenerateTokensForOptedOut()`. This applies `policy=1` in the [/token/generate](../endpoints/post-token-generate.md#token-generation-policy) call. Support for `policy=0` will be removed soon. (**GWH_SW03 same comment as earlier occurrence.**)
+   :::
+
+4. If the HTTP response status code is _not_ 200, see [Response Status Codes](../endpoints/post-token-generate.md#response-status-codes) to determine next steps. Otherwise, convert the UID2 identity response content into a `TokenGenerateResponse` object:
+
+   ```java
+   TokenGenerateResponse tokenGenerateResponse = publisherUid2Helper.createTokenGenerateResponse({response body}, envelope);
+   ```
+
+#### Standard Integration
+
+If you're using standard integration (client and server) (see [UID2 SDK for JavaScript Integration Guide](../guides/publisher-client-side.md)): (**GWH_SW04 should this link to the client-side doc?**)
```

Comment:
again it's the standard guide doc 
http://localhost:3006/docs/guides/integration-javascript-standard

Reply:
Done, two instances in this file and two instances in the repo readme, thx.